sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py

```diff
@@ -351,7 +351,7 @@ async def _initial_request(self):
                 self.size = self._file_size
 
         except HttpResponseError as error:
-            if self._start_range is None and error.response.status_code == 416:
```

Comment:
`response` is an optional field for `HttpResponseError`. When passed, it sets `self.status_code = self.response.status_code`.

https://github.com/Azure/azure-sdk-for-python/blob/a23a8f6f100b27629ee6f2c326ed66a2938bd908/sdk/core/azure-core/azure/core/exceptions.py#L292-L301

Reply:
and now we no longer get an attribute error when `error.response = None`

Reply:
Cool, makes sense.